lib/dams/controller_helper.rb

```diff
@@ -675,8 +675,12 @@ def audit( id = "unknown" )
     def referrer_controller( request )
       uri = URI(request.referrer || "")
       if uri.host == request.host
```

Comment:
I tried it earlier but some tests failed. I think this is because /dc is environment specific and our test environment don't have the /dc url pattern, so it failed to retrieve the controller from the referrer url.